datadog/cmd/main/main.go

```diff
@@ -213,6 +214,90 @@ func (d *DatadogCostSource) getDDCostsForWindow(window opencost.Window, listPric
 	// datadog's usage API sometimes provides usages that get counted multiple times
 	// this post processing stage de-duplicates those usages and costs
 	postProcess(&ccResp)
+
+	// query from the first of the window's month until the window end's day so that we can properly adjust for the
+	// cumulative nature of the response
+	startDate := time.Date(window.Start().Year(), window.Start().Month(), 1, 0, 0, 0, 0, time.UTC)
+	endDate := time.Date(window.End().Year(), window.End().Month(), window.End().Day(), 0, 0, 0, 0, time.UTC)
+
+	view := "sub-org"
+	params := datadogV2.NewGetEstimatedCostByOrgOptionalParameters()
+	params.StartDate = &startDate
+	params.EndDate = &endDate
+	params.View = &view
+	resp, r, err := d.usageApi.GetEstimatedCostByOrg(d.ddCtx, *params)
+	if err != nil {
+		log.Errorf("Error when calling `UsageMeteringApi.GetEstimatedCostByOrg`: %v\n", err)
+		log.Errorf("Full HTTP response: %v\n", r)
+		ccResp.Errors = append(ccResp.Errors, err.Error())
+	}
+
+	previousChargeCosts := make(map[string]float32)
+
+	// estimated costs from datadog are per-day, so we scale in the event that we want hourly costs
```

Comment:
Ah, I didn't know this. Does it make sense to only have daily costs, then, and not support hourly at all with this plugin? Or is that not an option?

Reply:
I'd love to do that, but it's not really an option with the current plugin design. I believe that uniformly distributing the costs across the whole day gives us a "good enough" solution.